File: php/features.php
<?php
declare(strict_types=1);

// BluefinTecsMerchantPortal SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/AuditFeature.php';
require_once __DIR__ . '/feature/ClienttrackFeature.php';
require_once __DIR__ . '/feature/IdempotencyFeature.php';
require_once __DIR__ . '/feature/LogFeature.php';
require_once __DIR__ . '/feature/MetricsFeature.php';
require_once __DIR__ . '/feature/PagingFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TelemetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class BluefinTecsMerchantPortalFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BluefinTecsMerchantPortalBaseFeature();
            case "audit":
                return new BluefinTecsMerchantPortalAuditFeature();
            case "clienttrack":
                return new BluefinTecsMerchantPortalClienttrackFeature();
            case "idempotency":
                return new BluefinTecsMerchantPortalIdempotencyFeature();
            case "log":
                return new BluefinTecsMerchantPortalLogFeature();
            case "metrics":
                return new BluefinTecsMerchantPortalMetricsFeature();
            case "paging":
                return new BluefinTecsMerchantPortalPagingFeature();
            case "ratelimit":
                return new BluefinTecsMerchantPortalRatelimitFeature();
            case "retry":
                return new BluefinTecsMerchantPortalRetryFeature();
            case "telemetry":
                return new BluefinTecsMerchantPortalTelemetryFeature();
            case "test":
                return new BluefinTecsMerchantPortalTestFeature();
            case "timeout":
                return new BluefinTecsMerchantPortalTimeoutFeature();
            default:
                return new BluefinTecsMerchantPortalBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "audit":
            case "clienttrack":
            case "idempotency":
            case "log":
            case "metrics":
            case "paging":
            case "ratelimit":
            case "retry":
            case "telemetry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}

File: rb/features.rb
# BluefinTecsMerchantPortal SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/audit_feature'
require_relative 'feature/clienttrack_feature'
require_relative 'feature/debug_feature'
require_relative 'feature/idempotency_feature'
require_relative 'feature/log_feature'
require_relative 'feature/metrics_feature'
require_relative 'feature/paging_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/telemetry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module BluefinTecsMerchantPortalFeatures
  def self.make_feature(name)
    case name
    when "base"
      BluefinTecsMerchantPortalBaseFeature.new
    when "audit"
      BluefinTecsMerchantPortalAuditFeature.new
    when "clienttrack"
      BluefinTecsMerchantPortalClienttrackFeature.new
    when "debug"
      BluefinTecsMerchantPortalDebugFeature.new
    when "idempotency"
      BluefinTecsMerchantPortalIdempotencyFeature.new
    when "log"
      BluefinTecsMerchantPortalLogFeature.new
    when "metrics"
      BluefinTecsMerchantPortalMetricsFeature.new
    when "paging"
      BluefinTecsMerchantPortalPagingFeature.new
    when "ratelimit"
      BluefinTecsMerchantPortalRatelimitFeature.new
    when "retry"
      BluefinTecsMerchantPortalRetryFeature.new
    when "telemetry"
      BluefinTecsMerchantPortalTelemetryFeature.new
    when "test"
      BluefinTecsMerchantPortalTestFeature.new
    when "timeout"
      BluefinTecsMerchantPortalTimeoutFeature.new
    else
      BluefinTecsMerchantPortalBaseFeature.new
    end
  end
end
